SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/common/CompilationCoordinator.swift

```diff
@@ -12,14 +12,55 @@
 
 import Foundation
 
-protocol BuildCommand {
-  var arguments: BuildArguments { get }
-}
-
-extension BuildCommand {
-  func buildProduct(productName: String) throws {
-    logSection("Building product " + productName)
-    try build(packageDir: Paths.packageDir, name: productName, isProduct: true)
+struct CompilationCoordinator {
+  /// The path to the toolchain that shall be used to build SwiftSyntax.
+  let toolchain: URL
+
+  /// The directory in which build products shall be put.
+  ///
+  /// If omitted a directory named ".build" will be put in the swift-syntax directory.
+  let buildDir: URL?
+
+  /// Path to an Xcode workspace to create a unified build of SwiftSyntax with other projects.
+  let multirootDataFile: URL?
+
+  /// Build in release mode.
+  let release: Bool
+
+  /// When constructing RawSyntax nodes validate that their layout matches that
+  /// defined in `CodeGeneration` and that TokenSyntax nodes have a `tokenKind`
+  /// matching the ones specified in `CodeGeneration`.
+  let enableRawSyntaxValidation: Bool
+
+  /// For each `assertParse` test, perform mutations of the test case based on
+  /// alternate token choices that the parser checks, validating that there are
+  /// no round-trip or assertion failures.
+  let enableTestFuzzing: Bool
+
+  /// Treat all warnings as errors.
+  let warningsAsErrors: Bool
+
+  /// Enable verbose logging.
+  let verbose: Bool
```

Comment:
Could most/all of these variables be private? That way it would be clearer what the designated entry points for the struct are.

Reply:
I think so. I can not see any reason why they could not be.